hyp3lib/ingest_S1_granule.py

```diff
@@ -1,65 +1,66 @@
-from __future__ import print_function, absolute_import, division, unicode_literals
-
-from hyp3lib.execute import execute
 import logging
+import os
 import shutil
-from hyp3lib.par_s1_slc_single import par_s1_slc_single
+
+from hyp3lib import ExecuteError, OrbitDownloadError
 from hyp3lib.SLC_copy_S1_fullSW import SLC_copy_S1_fullSW
+from hyp3lib.execute import execute
 from hyp3lib.getBursts import getBursts
-from hyp3lib.get_orb import downloadSentinelOrbitFile
-import os
+from hyp3lib.get_orb import download_sentinel_orbit_file
+from hyp3lib.par_s1_slc_single import par_s1_slc_single
 
 
-def ingest_S1_granule(inFile,pol,look_fact,outFile):
+def ingest_S1_granule(safe_dir: str, pol: str, looks: int, out_file: str, orbit_file: str = None):
+    """Pre-process S1 imagery into GAMMA format
 
+    Args:
+        safe_dir: Sentinel-1 SAFE directory location
+        pol: polarization (e.g., 'vv')
+        looks: the number of looks to take
+        out_file: file name of the output GAMMA formatted imagery
+        orbit_file: Orbit file to use (will download a matching orbit file if None)
+    """
     pol = pol.lower()
-    inputType = inFile[7:11]
-    grd = "{}.grd".format(pol)
-    
-    # Ingest the granule into gamma format
-    if "GRD" in inputType:
-        cmd = "par_S1_GRD {inf}/*/*{pol}*.tiff {inf}/*/*{pol}*.xml {inf}/*/*/calibration-*{pol}*.xml " \
-              "{inf}/*/*/noise-*{pol}*.xml {grd}.par {grd}".format(inf=inFile,pol=pol,grd=grd)
-        execute(cmd,uselogging=True)
+    granule_type = safe_dir[7:11]
```

Comment:
`[7:11]` will be one of `GRDH`, `GRDM`, `SLC_`, we could just grab `[7:10]` and use `if granule_type == 'GRD'` in the subsequent check